graphql_compiler/tests/integration_tests/test_backends_integration.py

```diff
@@ -426,6 +426,92 @@ def test_recurse(self) -> None:
                     {"descendant_name": "Animal 4"},  # depth 3 match
                 ],
             ),
+            # Query 8: Skip depth 0
+            (
+                """
+            {
+                Animal {
+                    name @filter(op_name: "=", value: ["$starting_animal_name"])
+                    out_Animal_ParentOf {
+                        out_Animal_ParentOf @recurse(depth: 2){
+                            name @output(out_name: "descendant_name")
+                        }
+                    }
+                }
+            }""",
+                [
+                    {"descendant_name": "Animal 1"},  # depth 1 match
+                    {"descendant_name": "Animal 2"},  # depth 1 match
+                    {"descendant_name": "Animal 3"},  # depth 1 match
+                    {"descendant_name": "Animal 1"},  # depth 2 match
+                    {"descendant_name": "Animal 2"},  # depth 2 match
+                    {"descendant_name": "Animal 3"},  # depth 2 match
+                    {"descendant_name": "Animal 4"},  # depth 2 match
+                    {"descendant_name": "Animal 1"},  # depth 3 match
+                    {"descendant_name": "Animal 2"},  # depth 3 match
+                    {"descendant_name": "Animal 3"},  # depth 3 match
+                    {"descendant_name": "Animal 4"},  # depth 3 match
+                ],
+            ),
+            # Query 9: Output child name
+            (
+                """
+            {
+                Animal {
+                    name @filter(op_name: "=", value: ["$starting_animal_name"])
+                    out_Animal_ParentOf {
+                        name @output(out_name: "child_name")
+                        out_Animal_ParentOf @recurse(depth: 2){
+                            name @output(out_name: "descendant_name")
+                        }
+                    }
+                }
+            }""",
+                [
+                    {"child_name": "Animal 1", "descendant_name": "Animal 1"},  # depth 1 match
+                    {"child_name": "Animal 2", "descendant_name": "Animal 2"},  # depth 1 match
+                    {"child_name": "Animal 3", "descendant_name": "Animal 3"},  # depth 1 match
```

Comment:
nit: off-by-one depth notation relative to the compiler's spec, these are depth-0 because the query says `@recurse(depth: 2)` i.e. the depths are 0 (no hops), 1 (one hop), and 2 (two hops), instead of 1-2-3 as the comments say here